argilla-server/src/argilla_server/api/handlers/v1/datasets/datasets.py

```diff
@@ -161,7 +161,23 @@ async def get_dataset_progress(
 
     await authorize(current_user, DatasetPolicy.get(dataset))
 
-    return await datasets.get_dataset_progress(db, dataset_id)
+    return await datasets.get_dataset_progress(db, dataset.id)
+
+
+@router.get(
+    "/datasets/{dataset_id}/progress/annotators", response_model=AnnotatorsProgress, response_model_exclude_unset=True
```

Comment:
The `/annotators` part is misleading because we are not filtering by user role right? 

If that's the case I think this endpoint should be named `/datasets/{dataset_id}/users/progress` or something similar, at the end that's what we are doing, returning the progress by user for a specific dataset.